packages/lib/src/components/Card/Card.test.tsx

```diff
@@ -128,6 +128,47 @@ describe('Card', () => {
         });
     });
 
+    describe('formatData', () => {
+        const i18n = global.i18n;
+        const resources = global.resources;
+        const srPanel = global.srPanel;
+
+        const coreProps = { loadingContext: 'test', i18n, modules: { resources, srPanel } };
+
+        test('should echo back holderName if is a stored card', () => {
+            const card = new CardElement({ ...coreProps, storedPaymentMethodId: 'xxx', holderName: 'Test Holder' });
+            render(card.render());
+
+            expect(card.formatData().paymentMethod.holderName).toContain('Test Holder');
+        });
+
+        test('should NOT echo back holderName from data if is a stored card', () => {
+            const card = new CardElement({ ...coreProps, storedPaymentMethodId: 'xxx', data: { holderName: 'Test Holder' } });
+            render(card.render());
+
+            expect(card.formatData().paymentMethod.holderName).toContain('');
+        });
+
+        test('if no holderName specificed and is stored card, holder name should be empty string', () => {
+            const card = new CardElement({ ...coreProps, storedPaymentMethodId: 'xxx' });
+
+            expect(card.formatData().paymentMethod.holderName).toContain('');
+        });
+
+        test('if no holderName specificed and is not stored card, holder name should be empty string', () => {
+            const card = new CardElement({ ...coreProps, storedPaymentMethodId: 'xxx' });
+
+            expect(card.formatData().paymentMethod.holderName).toContain('');
+        });
+
+        test('should NOT echo back holderName if is not a stored card', () => {
+            const card = new CardElement({ ...coreProps, holderName: 'Test Holder' });
+            render(card.render());
+
+            expect(card.formatData().paymentMethod.holderName).toBeUndefined();
+        });
```

Comment:
Should we add a test that if it is not a storedCard, but `holderName` is passed in a _data_ object, then we _should_ see that in the data?

Reply:
Yeah, I think that should be covered too.

Reply:
Tests added.